GenericBot/EventHandlers/MessageEventHandler.cs

```diff
@@ -70,6 +70,18 @@ public static async Task MessageRecieved(SocketMessage parameterMessage, bool ed
             {
                 var dbUser = Core.GetUserFromGuild(parameterMessage.Author.Id, parameterMessage.GetGuild().Id);
                 dbUser.IncrementPointsAndMessages();
+
+                var dbGuild = Core.GetGuildConfig(parameterMessage.GetGuild().Id);
+                if (dbGuild.TrustedRoleId != 0 && dbUser.Points > dbGuild.TrustedRolePointThreshold)
+                {
+                    var guild = Core.DiscordClient.GetGuild(dbGuild.Id);
+                    var guildUser = guild.GetUser(dbUser.Id);
+                    if (!guildUser.Roles.Any(sr => sr.Id == dbGuild.TrustedRoleId))
+                    {
+                        guildUser.AddRoleAsync(guild.GetRole(dbGuild.TrustedRoleId));
```

Comment:
I deliberately don't record the role in Mongo here. The idea is that, upon leaving, the role is not automagically restored if the user rejoins.